src/casp/cromwell/bq_ops/extract/extract_command.wdl

```diff
@@ -0,0 +1,59 @@
+task extract {
+    String version = "extract_v1.0.0"
+    String project_id
+    String bq_dataset
+    String extract_table_prefix
+    Int start_bin
+    Int end_bin
+    String output_bucket_name
+    String output_bucket_directory
+
+    command {
+        cd /app
+        echo "START EXTRACTING"
+        python casp/services/bq_ops/extract/main.py \
+        --dataset ${bq_dataset} \
+        --extract_table_prefix ${extract_table_prefix} \
+        --start_bin ${start_bin} \
+        --end_bin ${end_bin} \
+        --output_bucket_name ${output_bucket_name} \
+        --output_bucket_directory ${output_bucket_directory}
+    }
+
+    runtime {
+        docker: "us-central1-docker.pkg.dev/dsp-cell-annotation-service/cas-services-cicd/cas-pytorch:fg-extract-01"
+        bootDiskSizeGb: 20
+        memory: "20G"
+        cpu: 10
+        zones: "us-central1-a"
+        maxRetries: 0
+        preemptible_tries: 0
+    }
+}
+
+workflow CASExtractBQ {
+    String project_id
+    String bq_dataset
+    String extract_table_prefix
+    Array[Array[Int]] bin_borders
+    String output_bucket_name
+    String output_bucket_directory
+
+
+    meta {
+        description: "Extract Data from BigQuery to anndata files"
+    }
+
+    scatter(bin_border in bin_borders) {
+        call extract {
+            input:
```

Comment:
The neat trick is this... if you have inputs which can just be "passed through" from the workflow-input level to the task-input level, then you don't need to pass them from the workflow to the task.  So here you could just have
```wdl
input:
    start_bin = bin_border[0],
    end_bin = bin_border[1],
```
and leave out everything else.  What will happen then is that the other inputs need to be defined for the task itself, not for the workflow, so (if I'm not mistaken) you'd have to change your input JSON a little bit.  But in my opinion, sometimes this is better, since there's no chance of some kind of unintentional typo.

I think the input JSON would then look like this:
```json
{
   "CASExtractBQ.extract.project_id": "dsp-cell-annotation-service",
   "CASExtractBQ.extract.bq_dataset": "cas_benchmark_v1",
   "CASExtractBQ.extract.extract_table_prefix": "kc_bench_v1",
   "CASExtractBQ.bin_borders": [[0, 9], [10, 19], [20, 29], [30, 39], [40, 49], [50, 59], [60, 69], [70, 79], [80, 89], [90, 99]],
   "CASExtractBQ.extract.output_bucket_name": "dsp-cell-annotation-service",
   "CASExtractBQ.extract.output_bucket_directory": "cromwell_extract_test"
 }
```
so that the task name `extract` is specified to tell the input where it's going (for all except `bin_borders`).

Reply:
That really works, thanks!